env0/utils.go

```diff
@@ -41,31 +41,28 @@ func intPtr(i int) *int {
 	return &i
 }
 
-// Extracts values from the resourcedata, and writes it to the interface.
-func readResourceData(i interface{}, d *schema.ResourceData) error {
+// Prepends prefix to the fieldName.
+func readResourceDataEx(prefix string, i interface{}, d *schema.ResourceData) error {
 	// TODO: add a mechanism that returns an error if fields were set in the resourceData but not in the struct.
 	// Blocked by: https://github.com/hashicorp/terraform-plugin-sdk/issues/910
 
 	val := reflect.ValueOf(i).Elem()
 	for i := 0; i < val.NumField(); i++ {
-		fieldName := val.Type().Field(i).Name
-		// Assumes golang is CamalCase and Terraform is snake_case.
-		// This behavior can be overrided be used in the 'tfschema' tag.
-		fieldNameSC := toSnakeCase(fieldName)
-		if resFieldName, ok := val.Type().Field(i).Tag.Lookup("tfschema"); ok {
-			if resFieldName == "-" {
-				continue
-			}
+		parsedField := getFieldName(val.Type().Field(i))
+		if parsedField.skip {
+			continue
+		}
 
-			// 'resource' tag found. Override to tag value.
-			fieldNameSC = resFieldName
+		fieldName := parsedField.name
+		if prefix != "" {
+			fieldName = prefix + "." + fieldName
 		}
 
 		field := val.Field(i)
 
 		fieldType := field.Type()
```

Comment:
Not related to this pr - so you can ignore this comment:
`fieldType` is defined in line 63 but is being used at line 90.
It might be better to move it closer to its usage

Reply:
It's a good point. I'll review the code down the line and see if I can improve a few things.
It's all covered with unit tests, so should be safe to try and make changes.